stopcovid/dialog/models/events.py

```diff
@@ -239,6 +240,23 @@ def apply_to(self, dialog_state: DialogState) -> None:
         dialog_state.user_profile.opted_out = False
 
 
+class UserCreated(DialogEvent):
+    event_type: DialogEventType = DialogEventType.USER_CREATED
+    name: str
+    unit_id: int
+    employer_id: int
+
+    def apply_to(self, dialog_state: DialogState) -> None:
+        dialog_state.user_profile.name = self.name
+        if dialog_state.user_profile.account_info:
+            dialog_state.user_profile.account_info.unit_id = self.unit_id
+            dialog_state.user_profile.account_info.employer_id = self.employer_id
+        else:
+            dialog_state.user_profile.account_info = AccountInfo(
+                unit_id=self.unit_id, employer_id=self.employer_id
+            )
```

Comment:
What should happen if the user already exists?

Reply:
hmm good question. seems we'd probably want to overwrite their profile state with whatever we publish? but that could have other consequences in the dialog engine--like we'd probably want to kill their current drill & drill instance id

Reply:
but maybe that's actually a bad experience. if they're already training, it feels strange to just wipe their slate clean